web/src/main/java/org/mskcc/oncotree/api/OncotreeMappingsApi.java

```diff
@@ -0,0 +1,96 @@
+package org.mskcc.oncotree.api;
+
+import org.apache.log4j.Logger;
+import org.apache.commons.lang3.StringUtils;
+import org.mskcc.oncotree.crosswalk.CrosswalkRepository;
+import org.mskcc.oncotree.crosswalk.MSKConcept;
+import org.mskcc.oncotree.model.OncotreeMappingsResp;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class OncotreeMappingsApi {
+	
+    @Autowired
+    private CrosswalkRepository crosswalkRepository;
+	
+    private final static Logger logger = Logger.getLogger(OncotreeMappingsApi.class);
+	
+	@RequestMapping("api/crosswalk")
+	public OncotreeMappingsResp getMappings(
+										@RequestParam(value="vocabularyId", required=false) String vocabularyId,
+										@RequestParam(value="conceptId", required=false) String conceptId,
+										@RequestParam(value="histologyCode", required=false) String histologyCode,
+										@RequestParam(value="siteCode", required=false) String siteCode
+										) {
+		
+		if( !validateMappingParameters(vocabularyId,
+									   conceptId,
+									   histologyCode,
+									   siteCode)
+									  ){
+    		throw new RuntimeException("Your query parameters, vocabularyId: " + vocabularyId +
+    				", conceptId: " + conceptId + ", histologyCode: " + histologyCode +
+    				", siteCode: " + siteCode + " are not valid. Please refer to the documentation");
+		}
+		
+		MSKConcept mskConcept = crosswalkRepository.queryCVS(vocabularyId, conceptId, histologyCode, siteCode);
+		
+		return extractOncotreeMappings( mskConcept );
+	}
+	
+	private static boolean validateMappingParameters(
+													  String vocabularyId,
+													  String conceptId,
+													  String histologyCode,
+													  String siteCode
+													 ){
+		// vocabularyId can't be null
+		if(StringUtils.isEmpty(vocabularyId)){
+			return false;
+		}
+
+		// if there's a conceptId, both histology and site must be null

```

Comment:
ok ... although if the query can either be by conceptID or by histology/site, I think I would have preferred two different endpoints. But I guess we are duplicating the previous design.